### PR TITLE
Fixed Zod Validation #13

### LINKED ISSUE #13: Bug : Zod input schema validation fix required on input fields.
Hey contributor :wave: 
currently: In few place input validation is broken. [Empty Spaces OR White Spaces can be entered/updated]

1. Profile Page
image

2. Invite Page
image

3. Add Group Component
image

4. Add New Email Address Page
image

[please check in other places too]

Update Request : Show throw Errors in below the input fields OR Should show TOAST messages .

like this one below [ only on page based field (not on the components) ]
&
[on components just need to show error below input fields ]
image

### PR DESCRIPTION
**Pull request by skushagra9**
Added the trim() function to zod validation
Fixed #13

### CODE REVIEW

File: apps/web-app/pages/[orgSlug]/settings/user/profiles.vue
Comment (McPizza0) [inline apps/web-app/pages/[orgSlug]/settings/user/profiles.vue:1]: schema validation didnt exist here before
maybe we can just set it to z.string().trim() without a min/max

File: apps/web-app/pages/join/org.vue [hunk 1]
```diff
@@ -220,6 +220,7 @@
               .regex(/^[a-z0-9]*$/, {
                 message: 'Only letters and numbers'
               })
+              .trim()
```
Comment (McPizza0) [inline apps/web-app/pages/join/org.vue:223]: Same here, regex already covers, can remove .trim

File: apps/web-app/pages/join/index.vue [hunk 2]
```diff
@@ -108,6 +108,7 @@
             .regex(/^[a-zA-Z0-9]*$/, {
               message: 'Only letters and numbers'
             })
+            .trim()
```
Comment (McPizza0) [inline apps/web-app/pages/join/index.vue:111]: the regex here already covers spaces, we can remove .trim

File: apps/web-app/pages/join/profile.vue [hunk 3]
```diff
@@ -229,11 +229,13 @@
             v-model:value="fNameValue"
             v-model:valid="fNameValid"
             label="First Name"
+            :schema="z.string().min(1).max(32).trim()"
```
Comment (McPizza0) [inline apps/web-app/pages/join/profile.vue:232]: schema validation didnt exist here before
maybe we can just set it to z.string().trim() without a min/max

File: apps/web-app/pages/join/profile.vue [hunk 4]
```diff
@@ -229,11 +229,13 @@
...
             placeholder="" />
           <UnUiInput
             v-model:value="lNameValue"
             v-model:valid="lNameValid"
             label="Last Name"
+            :schema="z.string().min(1).max(32).trim()"
```
Comment (McPizza0) [inline apps/web-app/pages/join/profile.vue:238]: schema validation didnt exist here before
maybe we can just set it to z.string().trim() without a min/max

**Review by McPizza0 — CHANGES_REQUESTED** [review]
The order of operations matters with zod
z.string.min(1).max(3).trim() will return valid for a string of `' '`
the trim needs to come first, so all other rules are checked on the trimmed string

z.trim().min(1).max(5) will fail with a string of `' '`

can you please update all the schemas to reflect this

**Comment by skushagra9** [comment]
@McPizza0 Sure, I will make the changes

**Review by McPizza0 — APPROVED** [review]
lgtm

**Comment by McPizza0** [comment]
hey @skushagra9 

we have been strongly advised to add a contributor license agreement
motivation is to ensure that your contributions can only be "commercialized" or offered as a saas by the UnInbox team (Unproprietary Corporation) and not by others
please head over to https://cla-assistant.io/un/inbox and sign the agreement 
once we have more funds for legal stuff, we'll get a custom one built to add in clauses to the effect of "no changing to non-open licenses" etc.

thank you

**Comment by skushagra9** [comment]
sure, thanks

**Pull request merged**